src/openforms/utils/json_logic.py

```diff
@@ -49,3 +49,57 @@ def normalize_value(value: Union[dict, list, tuple, int, str]) -> JSONLogicValue
         return JsonLogicTest.from_expression(value)
 
     raise NotImplementedError(f"Unknown value type: {type(value)}")
+
+def find_component_key(test: JsonLogicTest, candidates: list) -> str:
+    for value in test.values:
+        if isinstance(value, JsonLogicTest) and value.operator == "var":
+            candidate = value.values[0]
+            if candidate in candidates:
+                return candidate
+
+        elif isinstance(value, JsonLogicTest) and value.operator == "date":
+            return find_component_key(value, candidates=candidates)
+
+    return ""
+
+
+def find_key_value(value: Union[JsonLogicTest, str]) -> str :
+    """Find the value associated to the key in the json_trigger_rule"""
+    if isinstance(value, JsonLogicTest) :
+        return find_key_value(value.values[0])
+
+    return value
+
+def formating_json_logic(logics :Union[dict, list, JsonLogicTest], candidates, submission):
```

Comment:
this function name should be something more accurate like `convert_json_logic_to_logging_tree` or something like that.